src/rest/server_service/error.rs

```diff
@@ -0,0 +1,43 @@
+use native_tls::Error as TlsError;
+use std::error::Error as StdError;
+use std::fmt::{Display, Formatter, Result as FmtResult};
+use std::io::Error as IoError;
+
+#[derive(Debug)]
+pub enum Error {
+    Pkcs12LoadFailed(IoError),
+    Pkcs12Invalid(TlsError),
+    BindFailed(IoError),
+    ServerAlreadyStopped,
+    ServerStopTimeout,
+    ServerStopFailed,
+}
+
+impl Display for Error {
+    fn fmt(&self, f: &mut Formatter) -> FmtResult {
+        let error_name = match *self {
+            Error::Pkcs12LoadFailed(_) => "Pkcs12LoadFailed",
+            Error::Pkcs12Invalid(_) => "Pkcs12Invalid",
+            Error::BindFailed(_) => "BindFailed",
+            Error::ServerAlreadyStopped => "ServerAlreadyStopped",
+            Error::ServerStopTimeout => "ServerStopTimeout ",
+            Error::ServerStopFailed => "ServerStopFailed ",
+        };
+        write!(f, "Server service error: {}", error_name)?;
+        if let Some(cause) = self.source() {
+            write!(f, "caused by {}", cause)?
```

Comment:
Don't do this here either, it's up to a top level error handler to walk the error chain if needed.

Reply:
Will fix it in next PR